src/ServerMain.cpp

```diff
@@ -144,9 +144,8 @@ int main(int argc, char** argv) {
     Server server(port, numThreads);
     server.initialize(index, text, allPermutations, usePatterns);
     server.run();
-  } catch (const ad_semsearch::Exception& e) {
-    LOG(ERROR) << e.getFullErrorMessage() << '\n';
-    return 1;
+  } catch (const std::exception& e) {
+    LOG(ERROR) << e.what() << std::endl;
   }
   return 0;
```

Comment:
Is there any way that we legally reach this `return 0` statement? We could remove it to suggest that it is never reached. In addition provably `server.run()` could get a ` [[noreturn]]` attribute if it is indeed an infinite loop that can be only left by throwing. (Unrelated to your changes, I just see this now.

Reply:
I tried marking it `[[noreturn]]` but the compiler thinks it does return because it uses `thread::join()` and the compiler can't see that the threads will only terminate with exceptions.